twill-api/src/main/java/org/apache/twill/api/TwillController.java

```diff
@@ -89,4 +90,30 @@
    * @return A {@link Future} that will be completed when the restart operation has been done.
    */
   Future<String> restartInstances(String runnable, int instanceId, int... moreInstanceIds);
+
+  /**
+   * Set the root log level for Twill applications in all containers.
+   *
+   * @param logLevel The log level for the root logger to change.
+   * @return A {@link Future} that will be completed when the set log level operation has been done.
```

Comment:
The `String` in `Future` in `setLogLevel` and `setLogLevels` for the application is the string representation of the log levels passed in, the `String` in `Future` for `setLogLevels` for a runnable is the runnableName.